test/attempts.test.js

```diff
@@ -23,4 +24,19 @@ describe('Request attempts', function () {
       done();
     });
   });
+
+  it('should call delay strategy 2 times after some retries', function (done) {
+    var mockDelayStrategy = sinon.stub().returns(100);
+
+    request({
+      url: 'http://www.filltext.com/?rows=1&err=500', // return a 500 status
+      maxAttempts: 3,
+      retryDelay: 100,
```

Comment:
> I looked through the code and retryDelay will not come into play if delayStrategy is a function. 

That's the goal indeed :)

> So even if we set it to a high number the test would still pass.

That was my point

> How about I add a check for the time between "before the request" and "the callback was called" is at least 200ms? For 3 attempts, delayStrategy is only called twice since the first call always goes through without delay.

Yep, that's was my last point :)

If you update your test, I will check it one last time and merge it :)